docs/Training-on-Amazon-Web-Service.md

```diff
@@ -1,53 +1,59 @@
 # Training on Amazon Web Service
 
-This page contains instructions for setting up an EC2 instance on Amazon Web Service for use in training ML-Agents environments. Current limitations of the Unity Engine require that a screen be available to render to. In order to make this possible when training on a remote server, a virtual screen is required. We can do this by installing Xorg and creating a virtual screen. Once installed and created, we can display the Unity environment in the virtual environment, and train as we would on a local machine. 
+This page contains instructions for setting up an EC2 instance on Amazon Web Service for use in training ML-Agents environments. You can run "headless" training if none of the agents in the environment use visual observations. 
 
 ## Pre-Configured AMI
 A public pre-configured AMI is available with the ID: `ami-30ec184a` in the `us-east-1` region. It was created as a modification of the Amazon Deep Learning [AMI](https://aws.amazon.com/marketplace/pp/B01M0AXXQB). 
 
 ## Configuring your own Instance
-Instructions here are adapted from this [Medium post](https://medium.com/towards-data-science/how-to-run-unity-on-amazon-cloud-or-without-monitor-3c10ce022639) on running general Unity applications in the cloud.
 
 1. To begin with, you will need an EC2 instance which contains the latest Nvidia drivers, CUDA8, and cuDNN.  There are a number of external tutorials which describe this, such as:
     * [Getting CUDA 8 to Work With openAI Gym on AWS and Compiling TensorFlow for CUDA 8 Compatibility](https://davidsanwald.github.io/2016/11/13/building-tensorflow-with-gpu-support.html)
     * [Installing TensorFlow on an AWS EC2 P2 GPU Instance](http://expressionflow.com/2016/10/09/installing-tensorflow-on-an-aws-ec2-p2-gpu-instance/)
     * [Updating Nvidia CUDA to 8.0.x in Ubuntu 16.04 – EC2 Gx instance](https://aichamp.wordpress.com/2016/11/09/updating-nvidia-cuda-to-8-0-x-in-ubuntu-16-04-ec2-gx-instance/)
-2. Move `python` to remote instance.
+
+## Installing ML-Agents
+
+2. Move `python` sub-folder of this repo to remote instance, and set as working directory.
 2. Install the required packages with `pip3 install .`.
-3. Run the following commands to install Xorg:
+
+## Testing
+
+If all steps worked correctly, upload a built Linux environment executable (with `headless` mode selected during build-time) to the instance, and test it from Python using:
```

Comment:
To verify that all steps worked correctly:

1. In the Unity Editor, load a project containing an ML-Agents environment (you can use one of the example environments if you have not created your own).
2. Open the **Build Settings** window (menu: **File** > **Build Settings**).  
3. Select **Linux** as the **Target Platform**.

[Editors note: does Architecture matter?]

4. Check **Headless Mode** (unless you have enabled a virtual screen following the instructions below).
5. Click **Build** to build the Unity environment executable.
6. Upload the executable to your EC2 instance.
7. Test the instance setup from Python using: